cli/command/utils.go

```diff
@@ -115,5 +117,23 @@ func PruneFilters(dockerCli Cli, pruneFilters filters.Args) filters.Args {
 		pruneFilters.Add(parts[0], parts[1])
 	}
 
+	// Server APIs older than 1.29 will just ignore label filters.
```

Comment:
Yes, that would be more correct, but the problem is that the issue is with old servers, so it's hard to fix now.

The server would have to error on all query params that it doesn't understand, which is a big change in behaviour.